File: app/src/main/java/com/seven/simpleandroid/extensions/ReflectEx.java
package com.seven.simpleandroid.extensions;

import androidx.annotation.NonNull;

import java.lang.reflect.Field;

public class ReflectEx {
    public static <T> void setField(T obj, @NonNull String field, Object value) throws NoSuchFieldException, IllegalAccessException {
        Class mClass = obj.getClass();
        Field mFiled = mClass.getDeclaredField(field);
        mFiled.setAccessible(true);
        mFiled.set(obj, value);
    }
    
    public static <V> V getField(Object obj, @NonNull String field) throws NoSuchFieldException, IllegalAccessException {
        Class mClass = obj.getClass();
        Field mFiled = mClass.getDeclaredField(field);
        mFiled.setAccessible(true);
        
        return (V) mFiled.get(obj);
    }
}

File: app/src/main/java/com/seven/simpleandroid/extensions/ReflectEx.kt
@file:Suppress("UNCHECKED_CAST")

package com.seven.simpleandroid.extensions

fun <T> Any.setField(mClass: Class<T>, field: String, value: Any) {
    val mField = mClass.getDeclaredField(field)
    mField.isAccessible = true
    mField.set(this, value)
}

fun Any.setField(field: String, value: Any) {
    setField(this::class.java, field, value)
}

fun <C, T> Any.getField(mClass: Class<C>, field: String) : T? {
    val mField = mClass.getDeclaredField(field)
    mField.isAccessible = true
    return mField.get(this) as T
}

fun <T> Any.getField(field: String) : T? {
    return getField(this::class.java, field)
}